### ISSUE TITLE
bug: overriding aria attributes is not easy or straightforward

### ISSUE THREAD

**Issue by BingSZou** [ionic-team/ionic-framework#20127]
# Bug Report

**Ionic version:**
[x] **4.x**

**Current behavior:**
Most Accessibility attributes, e.g. Aria-Label, Aria-LabeledBy, Role set on the Ion-Button does not work

**Expected behavior:**
The Accessibility attributes should work on Ion-Button

**Steps to reproduce:**
Create a simple Ion-Button with following attributes: 
```
<ion-button role="link" aria-label="this is my data">
    View my data </ion-button>
```
This sample code should be read out as:  "this is my data link",  but instead it reads out "View my data button"

**Other information:**
Root Cause:  Ion-Button has a shadow root: "button" . The attributes are not copied over to the button element.  So the aria attributes not working.

Note: A workaround is to create a directive with MutationObserver to copy the aria attributes from Ion-Button to its shadow-root button.

**Comment by VincentCrete** [comment]
@BingSZou, can-you show us an example of your workaround? We have the same problem.

**Comment by pschwindling** [comment]
Same problem here, which really is a big issue in our app now.

**Comment by VittorioEuris** [comment]
On Ionic@5.6.7 I see `ion-button` with `slot="icon-only"` apply the right aria-label but this is not updated on changes.
In details I'm using ngx-translate to provide i18n to my app but when user changes language dinamically inside the app every `ion-button` already instantiated doesn't update their value. This is a real problem because of buttons `slot="icon-only"` inside the header.

**Comment by sean-perkins** [comment]
Hello everyone 👋  with #25156, now all non-deprecated `aria-*` and `role` attributes are reflected to the element created within light DOM.

This change applies to:
- `ion-button`
- `ion-back-button`
- `ion-breadcrumb`
- `ion-header`
- `ion-input`
- `ion-textarea` 
- `ion-menu`
- `ion-menu-button`

Content this is slotted is already in the light DOM, so you should be able to already apply aria attributes to those elements directly.

If there is another component that needs this behavior, please let us know in a new issue. Thanks!
